apm-agent-core/src/main/java/co/elastic/apm/agent/bci/bytebuddy/AnnotationValueOffsetMappingFactory.java

```diff
@@ -48,18 +49,27 @@
         return new Advice.OffsetMapping() {
             @Override
             public Target resolve(TypeDescription instrumentedType, MethodDescription instrumentedMethod, Assigner assigner, Advice.ArgumentHandler argumentHandler, Sort sort) {
-                return Target.ForStackManipulation.of(getAnnotationValue(instrumentedMethod, annotation.loadSilent()));
+                return Target.ForStackManipulation.of(getAnnotationValue(instrumentedType, instrumentedMethod, annotation.loadSilent()));
             }
         };
     }
 
     @Nullable
-    private Object getAnnotationValue(MethodDescription instrumentedMethod, AnnotationValueExtractor annotationValueExtractor) {
-        for (TypeDescription typeDescription : instrumentedMethod.getDeclaredAnnotations().asTypeList()) {
+    private Object getAnnotationValue(TypeDescription instrumentedType, MethodDescription instrumentedMethod, AnnotationValueExtractor annotationValueExtractor) {
+        if (AnnotationType.CLASS.equals(annotationValueExtractor.type())) {
+            return getAnnotationValue(instrumentedType.getDeclaredAnnotations(), annotationValueExtractor);
+        } else {
+            return getAnnotationValue(instrumentedMethod.getDeclaredAnnotations(), annotationValueExtractor);
+        }
+    }
+
+    @Nullable
+    private Object getAnnotationValue(AnnotationList annotationList, AnnotationValueExtractor annotationValueExtractor) {
```

Comment:
@kananindzya solved it with https://github.com/elastic/apm-agent-java/pull/620/files#diff-ac2285d75eb737db8f38d5c2ff412ea7R87, so it can be used either way.